### PR TITLE
[api-minor] Extract the Content-Disposition filename

### PR DESCRIPTION
**Pull request by Snuffleupagus**
This PR includes the patch, rebased against the latest `master`, from PR #8688 (obviously maintaining correct author info), together with a follow-up patch to address some final comments to keep all `IPDFStreamReader` implementations consistent.

### CODE REVIEW

File: web/app.js [hunk 1]
```diff
@@ -1178,7 +1183,10 @@ let PDFViewerApplication = {
       }
 
       if (pdfTitle) {
-        this.setTitle(pdfTitle + ' - ' + document.title);
+        this.setTitle(
+          `${pdfTitle} - ${contentDispositionFilename || document.title}`);
```
Comment (yurydelendik) [inline web/app.js]: `document.title` contains HTML viewer title. I think we just want to see following instead this patch:

    if (!pdfTitle && contentDispositionFilename) pdfTitle = contentDispositionFilename;

**Review by yurydelendik — APPROVED** [review]
Looks good to me.

File: web/app.js (hunk 1 above)
Comment (yurydelendik) [reply web/app.js]: Actually, it's more complicated than I tought, having `document.title` to hold application variable is totally confusing. The change is good.

**Comment by timvandermeij** [comment]
Thank you!

**Pull request merged**